keras_cv/layers/preprocessing/random_flip_test.py

```diff
@@ -0,0 +1,137 @@
+            expected_output = inp
+            if mode == "horizontal" or mode == "horizontal_and_vertical":
+                expected_output = np.flip(expected_output, axis=2)
+            if mode == "vertical" or mode == "horizontal_and_vertical":
+                expected_output = np.flip(expected_output, axis=1)
+        with tf.compat.v1.test.mock.patch.object(
+            np.random,
+            "choice",
+            side_effect=mock_random,
+        ):
+            layer = RandomFlip(mode)
+            actual_output = layer(inp, training=True)
+            self.assertAllClose(expected_output, actual_output)
+
+    def test_random_flip(self):
+        modes = ["horizontal", "vertical", "horizontal_and_vertical"]
+        for mode in modes:
+            self._run_test(mode)
+
+    def test_random_flip_horizontal_half(self):
+        np.random.seed(1337)
+        mock_random = [True, False]
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = input_images.copy()
+        expected_output[0, :, :, :] = np.flip(input_images[0, :, :, :], axis=1)
+        self._run_test("horizontal", expected_output, mock_random)
+
+    def test_random_flip_vertical_half(self):
+        np.random.seed(1337)
+        mock_random = [True, False]
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = input_images.copy()
+        expected_output[0, :, :, :] = np.flip(input_images[0, :, :, :], axis=0)
+        self._run_test("vertical", expected_output, mock_random)
+
+    def test_random_flip_inference(self):
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = input_images
+        layer = RandomFlip()
+        actual_output = layer(input_images, training=False)
+        self.assertAllClose(expected_output, actual_output)
+
+    def test_random_flip_default(self):
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = np.flip(np.flip(input_images, axis=1), axis=2)
+        mock_random = [True, True, True, True]
+        with tf.compat.v1.test.mock.patch.object(
+            np.random,
+            "choice",
+            side_effect=mock_random,
+        ):
+            with self.cached_session():
+                layer = RandomFlip()
+                actual_output = layer(input_images, training=True)
+                self.assertAllClose(expected_output, actual_output)
+
+    def test_config_with_custom_name(self):
+        layer = RandomFlip(name="image_preproc")
+        config = layer.get_config()
+        layer_1 = RandomFlip.from_config(config)
+        self.assertEqual(layer_1.name, layer.name)
+
+    def test_random_flip_unbatched_image(self):
+        input_image = np.random.random((4, 4, 1)).astype(np.float32)
+        expected_output = np.flip(input_image, axis=0)
+        mock_random = [True, True, True, True]
+        with tf.compat.v1.test.mock.patch.object(
```

Comment:
can we replace this with a new unit test mocking feature?

@rchao or @qlzh727 may know of one?

Reply:
Can you try `unittest.mock.patch.object` and see if it works as you expected?

Reply:
Thanks Rick! I updated the import. However the issue seems to be from not being able to use tensor bool in if conditions.